model/jpa/src/main/java/org/keycloak/models/jpa/JpaClientProviderFactory.java

```diff
@@ -70,9 +70,6 @@ public void postInit(KeycloakSessionFactory factory) {
                     if (Objects.equals(attrUpdateEvent.getAttributeName(), RealmAttributes.ADMIN_PERMISSIONS_ENABLED) && Boolean.parseBoolean(attrUpdateEvent.getAttributeValue())) {
                         KeycloakSession keycloakSession = attrUpdateEvent.getKeycloakSession();
                         RealmModel realm = attrUpdateEvent.getRealm();
-
-                        if (realm.getAdminPermissionsClient() != null) return;
```

Comment:
Better handle how the feature is initialized within the `AdminPermissionSchema`. We may eventually change the behavior when disabling/enabling to re-create the authorization model. It should also keep all related code in a single place.